src/filesystem/impls/filer/BlobUtils.js

```diff
@@ -0,0 +1,93 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, URL */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Handlers = require("filesystem/impls/lib/handlers");
+    var Log = require("filesystem/impls/lib/log");
+
+    // BlobUtils provides an opportunistic cache for BLOB Object URLs
+    // which can be looked-up synchronously.
+
+    var Filer = require("thirdparty/filer/dist/filer");
+    var Path  = Filer.Path;
+
+    // 2-way cache for blob URL to path for looking up either way:
+    // * urls - paths keyed on blobUrls
+    // * blobs - blobUrls keyed on paths
+    var urls  = {};
+    var blobs = {};
```

Comment:
I know I am the one who named these, but I think they are confusing.  Should we change `urls` to `paths`?